pkg/proxy/net/packetio.go

```diff
@@ -262,10 +262,7 @@ func (p *PacketIO) Flush() error {
 }
 
 func (p *PacketIO) GracefulClose() error {
-	if p.conn != nil {
-		return p.conn.SetDeadline(time.Now())
-	}
-	return nil
+	return p.conn.SetDeadline(time.Now())
```

Comment:
It's possible that GracefulClose may be called after Close, so I added the judgement.

Reply:
OK, so it also ignored `ErrClosed`.